doc/development/contributing.rst

```diff
@@ -304,11 +304,11 @@ be reflected the next time you open a Python interpreter and ``import mne``
 Finally, we'll add a few dependencies that are not needed for running
 MNE-Python, but are needed for locally running our test suite::
 
-    $ pip install -r requirements_testing.txt
+    $ pip install -e .[test]
```

Comment:
behavior change: now install full install + full set up test dependencies

Reply:
I don't think we should advertise this variant actually because it will force people to install PyQt6. Conda users -- including those of our installers -- will have PyQt5

Reply:
@drammock this is also an issue with the monolithic `dev` -- we can't directly recommend using it because it uses `full` which in turn installs PyQt6

Reply:
Can you point me at a discussion of why this fractionation exists?  My naive question is why can't we harmonize what gets installed by the installer, conda-forge recipe and pip (i.e., can't we always install PyQt5 regardless of install route)

Reply:
PyQt6 isn't available on on conda forge yet. PySide6 doesn't support interactive updates well. We could use PyQt5 everywhere but PyQt6 should be better as it's newer and has more features etc (so why not use it and promote it where we can).

Reply:
> PyQt6 should be better as it's newer and has more features etc (so why not use it and promote it where we can).

But presumably we can't actually take advantage of the new features, since we maintain PyQt5 compatibility?  Or are we doing `try/except` or `hasattr` to triage? Or maybe there's a big performance boost that doesn't require user code changes?  [This table](https://www.qt.io/product/qt6/qt-5-15-vs-6-5-feature-comparison) suggests that the only major new thing in Qt6 is "shader tools" (which *does* sound like something useful to us) but maybe that's not the whole story.

sorry, I don't want to create roadblocks here, but I feel like I'm still not seeing the compelling argument to require pyqt6 *when possible* and pyqt5 otherwise, if that complicates our users' ability to transition from being a user to being a contributor.

Stepping back a bit, would it work to change the contribution guidelines to say "if you want to contribute, please create an empty venv (however you like) and then do `pip install mne[dev] --only-deps`" instead of recommending to use conda/mamba? Then people will get pyqt6 *and* they won't even need that "install extra testing deps" step later. WDYT?

Reply:
PyQt5 doesn't receive any updates or bug fixes. I've encountered text rendering bugs that will never be fixed, so I decided to support only PySide6 in MNELAB. PyQt6 should also work, but I'd really stay away from Qt5 stuff.

👍 for recommending pip over conda.

Reply:
> I'm still not seeing the compelling argument to require pyqt6 when possible and pyqt5 otherwise

Yeah we don't explicitly use any Qt6 features yet but it's more about general GUI library usability, features, etc. Qt6 came out almost 3 years ago so it's disappointing to recommend that people use Qt5. We have no choice for our installers at the moment, though, since PyQt6 isn't available there yet 😞 If we change `full` to install PyQt5 then people who use `pip`/`venv` (and previously used `requirements.txt`) will experience a bit of a step back.

> Stepping back a bit, would it work to change the contribution guidelines to say ...

We could recommend using a `venv` but I think that's a harder step for most people in terms of remembering to activate, relationship to conda envs, etc. So it depends a bit on how advanced/experienced our audience is for the `dev` part.

To keep things simple and easy for us (in terms of maintenance) and the less experienced devs (in terms of what they need to do), we could focus on supporting what's required for development assuming you're using our installers. In 1.6 at least these will fortunately come with a lot of what's needed:

https://github.com/mne-tools/mne-installers/blob/46a2a7b205cfdfbc1b6969f6436ee3229fedb9f7/recipes/mne-python/construct.yaml#L157

If we consider others with different sys configs (like those using pip venvs) as "advanced" devs, maybe it's okay to expect them to use pytest `SKIP` statements to see what they need when they need it. If this is acceptable, then recommending in our docs that people do `pip install -e .[test]` or `pip install -e .[test,docs]` might be sufficient and easier than trying to roll an all-encompassing `dev` option.

Reply:
Do we even still need conda? I mean, for texting and in our install instructions. 

I believe @cbrnr only uses pip to install everything these days, right?

I personally use conda every day still, but as I'm moving all my development into dev containers, I started using pipx and pip more for easier setup.

Reply:
> I believe @cbrnr only uses pip to install everything these days, right?

Yes, I've been using pip (and venv) exclusively for a long time now, and I haven't had any major problems. Note that I haven't really tested PyVista though, since I don't need it in my daily workflow. If we recommend VS Code, by default venvs are auto-activated, so I don't think this will be a huge issue.

Reply:
> We could recommend using a `venv` but I think that's a harder step for most people in terms of remembering to activate, relationship to conda envs, etc. So it depends a bit on how advanced/experienced our audience is for the `dev` part.

what I meant by the "however you like" in "please create an empty venv (however you like)" was that e.g., conda users could do 
```console
$ conda create -n mnedev pip
$ conda activate mnedev
$ pip install -e .[dev]
```
...so no need to require new knowledge of `venv` if they understand *at least one way of creating environments*. Or stated differently: "recommend that contributors install via pip in their dev environment even if they use conda to manage environments".

> To keep things simple and easy for us (in terms of maintenance) and the less experienced devs (in terms of what they need to do), we could focus on supporting what's required for development assuming you're using our installers.

-0.5 on this; the installers contain a LOT of stuff that is not needed for MNE dev work (tensorflow, plotly, etc).